### PR TITLE
Adding cone primitives.

### PR DESCRIPTION
**Pull request by bperseghetti**
# 🦟 Bug fix

## Summary
This helps add the missing cone geometry for primitive/basic parametric shapes:

conetopple
cone

And is also valuable for visualizations of emitters/source that typically have conic-based spread as seen in this acoustic attack on an IMU by showing the affected area:

drone_attack

Associated PRs:
- https://github.com/gazebosim/gz-gui/pull/621
- https://github.com/gazebosim/gz-math/pull/594
- https://github.com/gazebosim/gz-msgs/pull/442
- https://github.com/gazebosim/gz-physics/pull/639
- https://github.com/gazebosim/gz-rendering/pull/1003
- https://github.com/gazebosim/gz-sim/pull/2410
- https://github.com/gazebosim/sdformat/pull/1418

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by scpeters** [comment]
@osrf-jenkins run tests please

File: include/gz/sim/Primitives.hh [hunk 1]
```diff
@@ -67,7 +68,7 @@ namespace gz
     /// \brief Return an SDF string of one of the available primitive shape or
     /// light types.
     /// \param[in] _typeName Type name of the of shape or light to retrieve.
-    /// Must be one of: box, sphere, cylinder, capsule, ellipsoid, directional,
+    /// Must be one of: box, sphere, cylinder, cone, capsule, ellipsoid, directional,
     /// point, or spot.
```
Comment (scpeters) [inline include/gz/sim/Primitives.hh:72]:
```suggestion
    /// Must be one of: box, sphere, cylinder, cone, capsule, ellipsoid,
    /// directional, point, or spot.
```
reduce line length <= 80 characters

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
Thank you @bperseghetti , another great contribution!

@azeey or @scpeters do you mind to take a another look? CI is very green !

**Comment by scpeters** [comment]
> @azeey or @scpeters do you mind to take a another look? CI is very green !

I have suggested an update to our example worlds to include a cone in https://github.com/gazebosim/gz-sim/pull/2448

**Pull request merged**